### PR TITLE
DRIVERS-2648 Add AWS Secrets handling for serverless

### PR DESCRIPTION
**Pull request by blink1073**
Also adds credentials for `serverless_next`.

Tested with https://github.com/mongodb/mongo-python-driver/pull/1428 and https://github.com/mongodb/mongo-go-driver/pull/1473.

### CODE REVIEW

File: .evergreen/serverless/README.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+bash $DRIVERS_TOOLS/.evergreen/serverless/setup-secrets.sh
+```
+
+If targeting the dev version of Serverless, use:
+
+```bash
+$DRIVERS_TOOLS/.evergreen/serverless/setup-secrets.sh serverless-next
+```
+
+The full task group might look like:
```
Comment (NoahStapp) [inline .evergreen/serverless/README.md:23]: What is this referencing? Feels very out of nowhere when compared to the previous lines.

File: .evergreen/serverless/create-instance.sh [hunk 2]
```diff
@@ -22,6 +22,14 @@ set +o xtrace # Disable xtrace to ensure credentials aren't leaked
 #   SERVERLESS_URI            SRV connection string for newly created instance
 #   SERVERLESS_INSTANCE_NAME  Name of newly created instance (required for "get" and "delete" scripts)
 
+SCRIPT_DIR=$(dirname ${BASH_SOURCE[0]})
+. $SCRIPT_DIR/../handle-paths.sh
+
+# Ensure that secrets have already been set up.
+if [ -f "$SCRIPT_DIR/secrets-export.sh" ]; then
```
Comment (NoahStapp) [inline .evergreen/serverless/create-instance.sh:29]: What if this file doesn't exist?

File: .evergreen/serverless/setup-secrets.sh [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+#
+# setup-secrets.sh
+#
+# Set up serverless secrets using Drivers AWS Secrets Manager.
+# Writes a secrets-export.sh file to this folder.
+set -eux
+
+SCRIPT_DIR=$(dirname ${BASH_SOURCE[0]})
+. $SCRIPT_DIR/../handle-paths.sh
+pushd $SCRIPT_DIR
```
Comment (NoahStapp) [inline .evergreen/serverless/setup-secrets.sh:11]: Should there be a matching `popd` at the end of this script?

**Review by NoahStapp — CHANGES_REQUESTED** [review]

File: .evergreen/serverless/create-instance.sh (hunk 2 above)
Comment (blink1073) [reply .evergreen/serverless/create-instance.sh:29]: Legacy invocations won't have the file, the individual env var checks below will handle it.

File: .evergreen/serverless/setup-secrets.sh (hunk 3 above)
Comment (blink1073) [reply .evergreen/serverless/setup-secrets.sh:11]: Done

**Review by NoahStapp — APPROVED** [review]

**Pull request merged**